client/landing/gutenboarding/onboarding-block/vertical-background.tsx

```diff
@@ -14,53 +21,55 @@ export interface VerticalBackgroundProps {
 	onLoad: () => void;
 }
 
-const VerticalBackground: FunctionComponent< VerticalBackgroundProps > = ( { id, onLoad } ) => {
-	const [ imageUrl, setImageUrl ] = useState< string | null >( null );
+const VerticalBackground: FunctionComponent< VerticalBackgroundProps > = ( { onLoad } ) => {
+	const { siteVertical } = useSelect( select => select( STORE_KEY ).getState() );
+	const [ imageUrl, setImageUrl ] = useState< string | undefined >();
 
 	// Prefetch the default image.
 	useEffect( () => void ( new window.Image().src = defaultImageUrl ), [] );
 
 	useEffect( () => {
-		const preloadImage = new window.Image();
-		const failureHandler = () => {
-			setImageUrl( defaultImageUrl );
-			onLoad();
-		};
-		// We get an esmodule wrapping the url here
-		const successHandler = ( { default: url }: { default: string } ) => {
-			preloadImage;
-			preloadImage.onload = () => {
-				setImageUrl( url );
+		if ( siteVertical ) {
+			const preloadImage = new window.Image();
+			const failureHandler = () => {
+				setImageUrl( defaultImageUrl );
 				onLoad();
 			};
-			preloadImage.onerror = failureHandler;
-			preloadImage.src = url;
-		};
-		import(
-			/**
-			 * Set eager mode because these are just image _paths_.
-			 */
-			/* webpackMode: "eager" */
-			/* webpackInclude: /\.jpg$/ */
-			`../../../../static/images/verticals/${ id }.jpg`
-		).then( successHandler, failureHandler );
-		return () => {
-			preloadImage.onload = null;
-			preloadImage.onerror = null;
-		};
-	}, [ id, onLoad ] );
+			// We get an esmodule wrapping the url here
+			const successHandler = ( { default: url }: { default: string } ) => {
+				preloadImage.onload = () => {
+					setImageUrl( url );
+					onLoad();
+				};
+				preloadImage.onerror = failureHandler;
+				preloadImage.src = url;
+			};
+			import(
+				/**
+				 * Set eager mode because these are just image _paths_.
+				 */
+				/* webpackMode: "eager" */
+				/* webpackInclude: /\.jpg$/ */
+				`../../../../static/images/verticals/${ siteVertical.id }.jpg`
+			).then( successHandler, failureHandler );
+			return () => {
+				preloadImage.onload = null;
+				preloadImage.onerror = null;
+			};
+		}
+	}, [ siteVertical, onLoad ] );
 
 	return (
 		<SwitchTransition mode="in-out">
 			<CSSTransition
-				key={ imageUrl || 'empty' }
+				key={ imageUrl || 'none' }
 				timeout={ 750 }
 				classNames="onboarding-block__background-fade"
 			>
 				<div
-					className="onboarding-block__background"
+					className={ classnames( 'onboarding-block__background', { 'has-background': imageUrl } ) }
```

Comment:
There is already one `has-background` classname higher in DOM hierarchy (on `onboarding-block__acquire-intent` element). With this additional one, we are making syncing styles between content and background more difficult.
One first regression can be seen when clearing the selected vertical: https://cloudup.com/cy6BP0fXZRT

A possible alternative would be having the `VerticalBackground` component as a wrapper over the content so `onLoad={ () => setHasBackground( true ) }` callback and keeping background state in 2 components wouldn't be needed anymore.

Reply:
I'm glad you raised this 👍 

In the acquire intent step, the background depends on the media query for a dark background preference _and_ on whether a background is shown. This is so we can adjust text colors for when they're displayed over a darkened image.

Why do we only apply dark when we have a background? Having dark styling independent of the background image would be a better approach and eliminate the need for special handling in the background and dark preference case.

Adding just a little bit of dark preference handling complicates things without being a significant improvement for users.

**I'd like to remove dark preference handling until we have broader support (design) for it.**

What do you think?

Thanks for spotting the regression, there's an existing issue with the same case, just a different one. The dark preference is not respected but an image is still shown in the same situation:

![Screen Shot 2019-12-24 at 11 41 28](https://user-images.githubusercontent.com/841763/71409608-889e6380-2642-11ea-92a9-70e40f122895.png)

Reply:
Yes and yes. Adding partial dark-theme support complicates things for and doesn't provide a consistent experience for the users.

Current technical concerns:
* transitioning white -> dark bg color and dark -> white text using just opacity wouldn't be easy
* style syncing issues already discussed between content and background
* adding state and callbacks (everything around `has-background`) to support dark-theme seems to be fragile

User-facing accessibility issues:
* transition from light to dark theme (once selecting a vertical) and from dark to light (when deselecting a vertical or when advancing to a non-dark step in the onboarding) could be confusing 
* we don't have a toggle theme (light/dark) button yet. It should be provided when Gutenberg supports dark-theme

So I agree with ending the dark-theme experiment now and re-adding the styling and media query once it is supported at top-level. We can create an issue for that where we can include current findings. 
cc: @dwolfe